gammapy/modeling/models/temporal.py

```diff
@@ -472,79 +471,93 @@ def __str__(self):
-        ----------
-        time : array_like
-            Time since the ``reference`` time.
-        ext : int or str, optional, default: 0
-            Parameter passed to ~scipy.interpolate.InterpolatedUnivariateSpline
-            Controls the extrapolation mode for GTIs outside the range
-            0 or "extrapolate", return the extrapolated value.
-            1 or "zeros", return 0
-            2 or "raise", raise a ValueError
-            3 or "const", return the boundary value.
+        return cls(m, filename=filename, t_ref=t_ref)
 
+    @classmethod
+    def read(cls, filename, format="table"):
+        """Read a TemplateModel from disk
 
-        Returns
-        -------
-        norm : array_like
-            Norm at the given times.
-        """
-        return self._interpolator(time, ext=ext)
+        Parameters:
 
-    def integral(self, t_min, t_max):
-        """Evaluate the integrated flux within the given time intervals
+        filename : str
+            Name of file to read
+        format : str
+            Format of the input file.
+            either "table" or "map"
 
-        Parameters
-        ----------
-        t_min: `~astropy.time.Time`
-            Start times of observation
-        t_max: `~astropy.time.Time`
-            Stop times of observation
-        Returns
-        -------
-        norm: The model integrated flux
         """
+        filename = str(make_path(filename))
+        if format == "table":
+            table = Table.read(filename)
+            cls.from_table(table)
 
-        n1 = self._interpolator.antiderivative()(t_max.mjd)
-        n2 = self._interpolator.antiderivative()(t_min.mjd)
-        return u.Quantity(n1 - n2, "day") / self.time_sum(t_min, t_max)
+        elif format == "map":
+            m = RegionNDMap.read(filenames)
+            t_ref = m.meta["MJDREFI"]
+            return cls(m, filename=filename, t_ref=t_ref)
+
+        else:
+            raise ValueError("Not a valid format")
+
+    def to_table(self):
+        """Convert model to an astropy table"""
+        columns = self.map.geom.axes
+        table = Table(data=self.map.quantity, names=columns,
+                      meta=self.map.meta)
+        return table
+
+    def write(self, filename, format="table", overwrite=False):
+        """Write a model to disk as per the specified format"""
+
+        if self.filename is None:
+            raise IOError("Missing filename")
+
+        if format == "table":
+            table = self.to_table()
+            table.write(filename, overwrite=overwrite)
+
+        elif format == "map":
+            self.map.write(filename, overwrite=overwrite)
+
+        else:
+            raise ValueError("Not a valid format")
+
+    def evaluate(self, time, energy=None):
+        """Evaluate the model at given coordinates."""
+        coord = {"time": time.mjd}
+        if energy is not None:
+            coord["energy"] = energy
+
+        coords = self.create_coords(coord)
```

Comment:
Is this working? The `self.create_coords()` method does not seem to exist? In any case it should be part of the non-public API, so please rename to `self._create_coords()`. But I'm not sure the method is even needed it seems it should just work to pass the coordinates to `self.map.interp_by_coord` like you do.